src/libostree/ostree-repo-pull.c

```diff
@@ -5695,7 +5695,7 @@ ostree_repo_resolve_keyring_for_collection (OstreeRepo    *self,
     }
 
   if (keyring_remote != NULL)
-    return ostree_remote_ref (keyring_remote);
+    return keyring_remote;
```

Comment:
Let's use autoptr on the variable still, and then `g_steal_pointer()` here?  We gain consistency and keep the movement towards Rust-like "move semantics".

Reply:
good point. I pushed another change in the fixup too, so see if that looks good. It's a behavior change but I think in the right direction